packages/settingeditor/src/SettingsFormEditor.tsx

```diff
@@ -225,6 +226,11 @@ export class SettingsFormEditor extends React.Component<
     const trans = this.props.translator.load('jupyterlab');
     const icon = this.props.isCollapsed ? caretRightIcon : caretDownIcon;
 
+    const renderers: { [id: string]: Field } = {};
+    for (let id in this.props.components) {
+      renderers[id] = this.props.components[id].fieldRenderer!;
```

Comment:
As `this.props.components[id]` may be defined but only with a widget renderer, I would be more caution here:

```suggestion
      const renderer = this.props.components[id].fieldRenderer;
      if (renderer) {
        renderers[id] = renderer;
      }
```

At some point it will be interesting to add support for custom widget in settings editor too. But let's keep that out of this PR.